packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -785,6 +773,32 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
     // we don't have event for it, but want to keep the event handling interface, event is empty.
     _.invoke(this.props, 'onSelectedChange', {}, { ...this.props, value })
   }
+
+  private tryFocusTriggerButton = () => {
+    if (!this.props.search) {
+      this.buttonRef.current.focus()
+    }
+  }
+
+  private tryFocusSearchInput = () => {
+    if (this.props.search) {
+      this.inputRef.current.focus()
+    }
+  }
```

Comment:
Sorry, but I don't like these imperative methods, we have the same issues with SUIR's `Dropdown`, it's very hard to understand when something will be focus and why.

It will be great to introduce the declarative focus control to Stardust, something like this: https://github.com/pirelenito/react-controlled-focus

It can be out of the scope of this PR, but we should avoid the imperative stuff like this.

Reply:
I totally support this concern, this is extremely prone to bugs and regressions.

At the same time, I think we're nowhere near ready to make this kind of change for this PR, any suggestions that I can apply for this one?

Reply:
For the current PR I don't have any suggestions about this :(

Reply:
opened #853 and resolved this comment :(